test/memory-usage.js

```diff
@@ -1,29 +1,63 @@
-//test for memory leak after closing connections
+// Test for memory leak after closing connections.
 var ConnectionPool = require('../lib/connection-pool');
 var fs = require('fs');
 
-var count = 0;
-var mem = [];
+var connectionCount = 0;
+var currentGroup = 0;
+var memGroupsBytes = [];
+
 var groupCount = 20;
 var poolSize = 1000;
+var maxConnections = groupCount * poolSize;
+
+// Print initial memory usage at program start.
+printMemoryUsage(0, process.memoryUsage().heapUsed, 1);
```

Comment:
I don't think we shouldn't display the heap size when there are 0 failed connections, since we aren't using it in the leak detection